### ISSUE TITLE
Remove non-linked PF collections from HI reco and review the thresholds for reduced RecHits

### ISSUE THREAD

**Issue by perrotta** [cms-sw/cmssw#20072]
Pull request #19927 temporarily stores both the "Tmp" and the final PF candidates in the HI reco collections, thus significantly increasing the event size. The plan is to eventually drop the Tmp collections, but they would like to keep both temporarily for development related studies and  comparisons.

Since no significant HI reco production is foreseen for this year, it may make sense to temporarily store all those collections, provided they don't forget to remove them when not needed any more: this issue is meant to remind HI reco developers that those collections must be removed as soon as not necessary for the development related studies.

Moreover, since the reducedECALRecHit's are the modules which produce the most of additional event size in those HI reco events (https://github.com/cms-sw/cmssw/pull/19927#issuecomment-320280634) and consume an important fraction of the CPU (https://github.com/cms-sw/cmssw/pull/19927#issuecomment-320280028)  probably also the thresholds set for those reduced recHits should be reviewed and, if possible, optimized. Also this threshold review should get tracked in this issue.

@mandrenguyen

**Comment by perrotta** [comment]
assign reco

**Comment by perrotta** [comment]
assign reconstruction

**Comment by mandrenguyen** [comment]
@cfmcginn @kurtejung @mverwe @ttrk

**Comment by mandrenguyen** [comment]
@perrotta Can you summarize what the reduced rechit collections are intended for?  Previously we didn't even produce these collections at all.  I included it to silence some DQM/validation warnings.

**Comment by slava77** [comment]
On 8/7/17 6:56 AM, Matthew Nguyen wrote:
> @perrotta <https://github.com/perrotta> Can you summarize what the
> reduced rechit collections are intended for? Previously we didn't even
> produce these collections at all. I included it to silence some
> DQM/validation warnings.

most of the ecal reduced rechits are saved to provide full access to the
hits for egamma objects
to be able to recompute cluster shapes or use the info in object
recalibration.
The approach to save the hits works out as long as the fake rate of
egamma objects is reasonably low or if their number per event is small
enough.
Perhaps the default thresholds from pp are not applicable to HI.

**Comment by smuzaffar** [comment]
@perrotta , is this still an open issue?

**Comment by perrotta** [comment]
> @perrotta , is this still an open issue?

Uhm, probably no... I still see the tmp sequences included in RecoHI/Configuration/python/Reconstruction_hiPF_cff.py
@mandrenguyen are those sequences and collections still needed in HI PF reco, in which case this issue can get closed? Or can they get cleaned out, instead, to reduce the size of the HI reco outputs?
